frontend/src/components/Schedule/Schedule.tsx

```diff
@@ -127,38 +74,69 @@ const Schedule: FunctionComponent = () => {
     });
   };
 
+  const onEventDrop = ({ start, end, event, resourceId }: any) => {
+    const nextEvents = events.map((old) => (old.id === event.id
+      ? { ...old, start: new Date(start), end: new Date(end), resourceId }
+      : old));
+    setEvents(nextEvents);
+  };
+
+  // eslint-disable-next-line no-alert
+  const onSelectEvent = (event: any) => alert(event.title);
+
+  const onSelectSlot = ({ start, end, resourceId }: any) => {
+    // eslint-disable-next-line no-alert
+    const title = window.prompt('New Ride name');
+    if (title) {
+      const newEvent: CalEvent = {
+        start: new Date(start),
+        end: new Date(end),
+        title,
+        id: tempId,
+        resourceId,
+      };
+      setTempId(tempId + 1);
+      setEvents([...events, newEvent]);
+    }
+  };
+
+  const okHr = (hr: number) => hr === curStart.getHours();
+
   return (
     <>
       <h1 className={styles.heading}>Home</h1>
       <div className={styles.calendar_container}>
         <div className={styles.left}>
-          <Calendar
-            formats={{
-              timeGutterFormat: 'h A',
-            }}
+          <DnDCalendar
+            formats={{ timeGutterFormat: 'h A' }}
             localizer={localizer}
             events={filterEvents(events)}
             toolbar={false}
-            step={60}
-            timeslots={1}
+            step={5}
+            timeslots={12}
             showMultiDayTimes={true}
             defaultView="day"
+            onEventDrop={onEventDrop}
```

Comment:
rn if you drag the events, some of them would be resized instead of being dragged to somewhere. Also some other events cannot be resized. I think there's this `onEventResize` that you can implement to address this problem